app/Http/Controllers/Admin/Users/UserController.php

```diff
@@ -0,0 +1,200 @@
+<?php
+/**
+ * @author    MyBB Group
+ * @version   2.0.0
+ * @package   mybb/core
+ * @license   http://www.mybb.com/licenses/bsd3 BSD-3
+ */
+
+namespace MyBB\Core\Http\Controllers\Admin\Users;
+
+use DaveJamesMiller\Breadcrumbs\Manager as Breadcrumbs;
+use Illuminate\Http\Request;
+use Illuminate\Validation\Factory;
+use MyBB\Core\Database\Repositories\UserRepositoryInterface;
+use MyBB\Core\Database\Repositories\RoleRepositoryInterface;
+use MyBB\Core\Http\Controllers\Admin\AdminController;
+use MyBB\Core\Http\Requests\User\SaveUserRequest;
+
+class UserController extends AdminController
+{
+    /**
+     * @var Breadcrumbs
+     */
+    private $breadcrumbs;
+    
+    /**
+     * @var UserRepositoryInterface
+     */
+    private $userRepository;
+    
+    /**
+     * @var Factory $validator
+     */
+    private $validator;
+
+    /**
+     * @var Request $request
+     */
+    private $request;
+    
+    /**
+     * Create a new user controller instance.
+     *
+     * @param Breadcrumbs $breadcrumbs
+     * @param Factory $validator
+     * @param Request $request
+     * @param UserRepositoryInterface $userRepository
+     * @param RoleRepositoryInterface $roleRepository
+     */
+    public function __construct(
+        Breadcrumbs $breadcrumbs,
+        Factory $validator,
+        UserRepositoryInterface $userRepository,
+        RoleRepositoryInterface $roleRepository
+    ) {
+        $this->breadcrumbs = $breadcrumbs;
+        $this->userRepository = $userRepository;
+        $this->validator = $validator;
+        $this->userRepository = $userRepository;
+        $this->roleRepository = $roleRepository;
+    }
+    
+    /**
+     * Show the user list.
+     *
+     * @return \Illuminate\View\View
+     */
+    public function users()
+    {
+        $this->breadcrumbs->setCurrentRoute('admin.users.list');
+        $users = $this->userRepository->all();
+
+        return view('admin.users.list', compact('users'))->withActive("users");
+    }
+    
+    /**
+     * Show the user edit form.
+     *
+     * @param int $id
+     *
+     * @return \Illuminate\View\View
+     */
+    public function editUser($id)
+    {
+        $this->breadcrumbs->setCurrentRoute('admin.users.edit');
+        $user = $this->userRepository->find($id);
+        $role = $user->displayRole();
+        $roles = $this->roles();
+        
+        return view('admin.users.edit', compact('user', 'roles', 'role'))->withActive("users");
+    }
+    
+    /**
+     * Handle a user edit request.
+     *
+     * @param Request $request
+     * @param int $id
+     *
+     * @return \Illuminate\Http\RedirectResponse
+     */
+    public function saveUser(Request $request, $id)
```

Comment:
Hi,

Normally we'd create a request type to handle validation, but I can do that later :)